lib/functions.php

```diff
@@ -551,7 +551,7 @@ function elgg_solr_add_update_object_default($entity) {
 	$doc->enabled = $entity->enabled;
 
 	$params = array('entity' => $entity);
-	$doc = elgg_trigger_plugin_hook('elgg_solr:index', $entity->type, $params, $doc);
+	$doc = elgg_trigger_plugin_hook('elgg_solr:index', $entity->type . (($subtype != '') ? ':' . $subtype : ''), $params, $doc);
```

Comment:
With this there will now be more specific hooks right?  So instead of 'elgg_solr:index', 'object' I would then hook into 'elgg_solr:index', 'object:blog'.  I like that, it saves an entity check on each handler, but it's a breaking change (any existing helper plugins would stop working if they updated to this right away).

I like it, but would we should create a new major tag for it if we go with it.